### DISCUSSION TITLE
Get default value of dirty field...

### DISCUSSION THREAD

**Discussion by abdullah-sameh** [react-hook-form/react-hook-form#11626] [Q&A]
i got the changed value, then i need to get the value before changed?

**Comment by Moshyfawn** [comment]
```js
const { defaultValues } = formState
```

**Reply by abdullah-sameh** [reply]
i need to get the defaultValues of my dirty fields in my form....

**Reply by Moshyfawn** [reply]
There isn't a state that specifically tracks dirty field defaults, but you can derive the value by combining the `dirtyFields` and `defaultValues` states. 

Something along the lines of

```js
const dirtyFieldsDefaultValues = Object.keys(dirtyFields).reduce((acc, key) => {
  if (defaultValues[key]) {
    acc[key] = defaultValues[key];
  }
  return acc;
}, {});
```
